### DISCUSSION TITLE
Questions regarding Confidential Computing

### DISCUSSION THREAD

**Discussion by JoongunPark** [NVIDIA/open-gpu-kernel-modules#702] [Q&A] [answered]
Hi, I'm looking for the source code or documentation related to Confidential VM Key Management in Multi-GPU Scenarios.

Specifically, I’m curious about how the confidential VM handles or mediates encryption keys between multiple GPUs.

My current understanding is that inter-GPU communication happens over NVLink, and the GPUs use AES-GCM-256 for message encryption. Could someone point me to the relevant part of the source code or any documentation that explains this key management process?

Thanks!

**Comment by VicoinDub** [comment, marked as answer]
Here are some related code comments—are these what you need?
https://github.com/NVIDIA/open-gpu-kernel-modules/blob/535.98/src/nvidia/src/kernel/gpu/conf_compute/arch/hopper/conf_compute_keystore_gh100.c#L33-L48C82

**Reply by JoongunPark** [reply]
Yes it seems so. Thank you so much! 
I will take a look more and close this comments after that!
